File: CSVExplorerJava/app/src/main/java/com/example/csvexplorerjava/entity/HeadersStore.java
package com.example.csvexplorerjava.entity;

import android.content.Context;
import android.content.SharedPreferences;

import org.json.JSONArray;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public final class HeadersStore {
    private static final String PREF = "dynamic_csv_prefs";
    private static final String KEY_HEADERS = "headers_json";

    private HeadersStore() {}

    public static void save(Context context, List<String> headers) {
        JSONArray arr = new JSONArray();
        if (headers != null) {
            for (String h : headers) arr.put(h);
        }
        SharedPreferences sp = context.getSharedPreferences(PREF, Context.MODE_PRIVATE);
        sp.edit().putString(KEY_HEADERS, arr.toString()).apply();
    }

    public static List<String> load(Context context) {
        SharedPreferences sp = context.getSharedPreferences(PREF, Context.MODE_PRIVATE);
        String s = sp.getString(KEY_HEADERS, null);
        if (s == null) return Collections.emptyList();

        try {
            JSONArray arr = new JSONArray(s);
            ArrayList<String> out = new ArrayList<>();
            for (int i = 0; i < arr.length(); i++) out.add(arr.getString(i));
            return out;
        } catch (Exception e) {
            return Collections.emptyList();
        }
    }

    public static void clear(Context context) {
        SharedPreferences sp = context.getSharedPreferences(PREF, Context.MODE_PRIVATE);
        sp.edit().remove(KEY_HEADERS).apply();
    }
}

File: CSVExplorer/app/src/main/java/com/example/csvexplorer/entity/HeadersStore.kt
package com.example.csvexplorer.entity

import android.content.Context
import org.json.JSONArray

object HeadersStore {
    private const val PREF = "dynamic_csv_prefs"
    private const val KEY_HEADERS = "headers_json"

    fun save(context: Context, headers: List<String>) {
        val arr = JSONArray()
        headers.forEach { arr.put(it) }
        context.getSharedPreferences(PREF, Context.MODE_PRIVATE)
            .edit()
            .putString(KEY_HEADERS, arr.toString())
            .apply()
    }

    fun load(context: Context): List<String> {
        val s = context.getSharedPreferences(PREF, Context.MODE_PRIVATE)
            .getString(KEY_HEADERS, null) ?: return emptyList()

        return try {
            val arr = JSONArray(s)
            val out = ArrayList<String>()
            for (i in 0 until arr.length()) out.add(arr.getString(i))
            out
        } catch (_: Exception) {
            emptyList()
        }
    }

    fun clear(context: Context) {
        context.getSharedPreferences(PREF, Context.MODE_PRIVATE)
            .edit()
            .remove(KEY_HEADERS)
            .apply()
    }
}